homie-controller/src/types.rs

```diff
@@ -219,6 +220,38 @@ impl Node {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Extension {
+    pub id: String,
+    pub version: String,
+    pub homie_versions: Vec<String>,
+}
+
+/// An error which can be returned when parsing an `Extension` from a string.
+#[derive(Error, Debug)]
+#[error("Invalid extension '{0}'")]
+pub struct ParseExtensionError(String);
+
+impl FromStr for Extension {
+    type Err = ParseExtensionError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<_> = s.split(":").collect();
+        if let [id, version, homie_versions] = parts.as_slice() {
+            if let Some(homie_versions) = homie_versions.strip_prefix("[") {
+                if let Some(homie_versions) = homie_versions.strip_suffix("]") {
+                    return Ok(Extension {
+                        id: (*id).to_owned(),
+                        version: (*version).to_owned(),
+                        homie_versions: homie_versions.split(";").map(|p| p.to_owned()).collect(),
```

Comment:
It feels like this wants a couple of tests.

Reply:
Added.